liteflow-labs/liteflow-js
# NumberInput validation

### Project organization

- Related to https://github.com/liteflow-labs/marketplace-template/issues/72
- Related https://trello.com/c/eErTVlGD

MARKETPLACE TEMPLATE PR:
https://github.com/liteflow-labs/marketplace-template/pull/89/

### Description

Adding validation for number inputs throughout the library

### How to test

This is currently partially testable in storybook, some are gated behind the login flow and can be tested on this PR https://github.com/liteflow-labs/marketplace-template/pull/89/ : 

Components should be testable They are: 
- [ ] Offer -> Form -> Bid -> Default
- [ ] Offer -> Checkout -> Multiple
- [ ] Sales -> Auction -> Form -> Default
- [ ] Sales -> Direct -> Form -> Is Single / Is Multiple
- [ ] Token -> Form -> Create -> Is Single / Is Multiple

Templates (doesn't build right now therefore cannot be tested). Validate functions should be okay but I added the display of the errors in the UI and since I can't preview it, it's not tested right now. This could be tested in `marketplace-template` repo with PR https://github.com/liteflow-labs/marketplace-template/pull/89/ :
- [ ] Explore page -> Filters -> Min/Max price

To test for price inputs: 
- [ ] Value need to be positive
- [ ] Value can't exceed maximum decimal of currency
- [ ] Value cannot be empty unless optional

To test for quantity inputs:

- [ ] Value need to be positive
- [ ] Value need to be a minimum of 1
- [ ] Value can't exceed the maximum available quantity
- [ ] Value need to be integers

To test for royalty input:

- [ ] Value can be optional or 0
- [ ] Value need to be positive
- [ ] Value cannot exceed the maximum allowed royalty
- [ ] Is there a maximum decimal ? not for now.

### Checklist
